src/osd/recovery_types.h

```diff
@@ -49,6 +85,11 @@ struct BackfillInterval {
     return end.is_max();
   }
 
+  /// true if begin exists in objects
+  bool is_valid_begin() const {
```

Comment:
How about is_begin_trimmed()?  begin is *valid* even if not present in objects.

Reply:
Changed to is_existing_begin